x/cosmwasmpool/genesis.go

```diff
@@ -1,20 +1,54 @@
 package cosmwasmpool
 
 import (
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/osmosis-labs/osmosis/v16/x/cosmwasmpool/types"
 )
 
 // InitGenesis initializes the store state from a genesis state.
-func (k *Keeper) InitGenesis(ctx sdk.Context, gen *types.GenesisState) {
+func (k *Keeper) InitGenesis(ctx sdk.Context, gen *types.GenesisState, unpacker codectypes.AnyUnpacker) {
 	k.SetParams(ctx, gen.Params)
+
+	// Add each genesis state pool to the x/cosmwasmpool module's state
+	for _, any := range gen.Pools {
+		var pool types.CosmWasmExtension
+		err := unpacker.UnpackAny(any, &pool)
+		if err != nil {
+			panic(err)
+		}
+		pool.SetWasmKeeper(k.wasmKeeper)
```

Comment:
Actually, why is this needed? We should be setting the store model to state, not the wrapped that has `wasmKeeper`

Reply:
We get the following error in the initiGenesis test when querying the spreadFactor without this line:

```
    --- FAIL: TestPoolModuleSuite/TestInitGenesis (0.40s)
        /Users/czar/Documents/osmosis/x/cosmwasmpool/suite.go:87: test panicked: kv store with key KVStoreKey{0x140010fbdc0, wasm} has not been registered in stores
            goroutine 56 [running]:
            runtime/debug.Stack()
            	/usr/local/go/src/runtime/debug/stack.go:24 +0x64
            github.com/stretchr/testify/suite.failOnPanic(0x140000ca4e0, {0x1021dbe80, 0x14000eaddd0})
            	/Users/czar/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:87 +0x34
            github.com/stretchr/testify/suite.Run.func1.1()
            	/Users/czar/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:183 +0x1f4
            panic({0x1021dbe80, 0x14000eaddd0})
            	/usr/local/go/src/runtime/panic.go:884 +0x204
            github.com/cosmos/cosmos-sdk/store/cachemulti.Store.GetKVStore({{0x1025b86b0, 0x14000f0d530}, 0x14000f0d5c0, 0x14000f22450, {0x0, 0x0}, 0x0, 0x14000f224e0}, {0x102588708, 0x140010fbdc0})
```

Reply:
Added your recommendation which fixed it here https://github.com/osmosis-labs/osmosis/pull/5696/commits/a4bafb5d258769bf7eb6f871c7ec3de6dd0d8430